mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -576,8 +609,7 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     (splits, categoricalFeaturesInfo_, labelType)
   }
 
-  override def runTest(rdd: RDD[LabeledPoint])
-    : Either[RandomForestModel, GradientBoostedTreesModel] = {
+  override def runTest(rdd: RDD[LabeledPoint]): DecisionTreeModel = {
```

Comment:
Agree with caching; but won't materializing them introduce an unnecessary pass since they are materialized on the first call to `model.transform` anyways?